Source/dom/components/ShareableObject.js

```diff
@@ -0,0 +1,73 @@
+import { DefinedPropertiesKey, WrappedObject } from '../WrappedObject'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapNativeObject } from '../wrapNativeObject'
+import { getDocumentData } from '../utils'
+
+const ObjectType = {
+  Symbol: 0,
+  LayerStyle: 1,
+  TextStyle: 2,
+  Unknown: 3,
+}
+
+export class ShareableObject extends WrappedObject {
```

Comment:
I'm not sure, but this might cause confusion. We have an `MSShareableObject` typedef in Sketch which is basically an `MSModelObject` that can be shared (e.g. a style or a symbol). The `MSShareableObjectReference` which this class appears to be wrapping ties a sharable object to a library.

Reply:
I'm guessing it's only confusing when working all day with `MSShareableObject` and `MSShareableObjectReference`? ;)

Reply:
Maybe - but it might be if people are doing class dumps etc.

What about calling it `SymbolReference`?

`MSShareableObjectReference` is a base class which could either be a `MSSymbolMasterReference` or a `MSSharedLayerReference` or a `MSSharedTextReference` (those last two should probably be renamed `MSSharedLayerStyleReference` and `MSSharedTextStyleReference` 😬 ). `MSShareableObjectReference` is useful for us as a generic way to deal with foreign objects, but for the API maybe it should be clear that this is a "symbol" object.

Reply:
I don't want to multiply the classes, it makes it super confusing and worrisome when looking at the documentation. How about `ImportableObject` instead?

Reply:
`ImportableObject` would work 👍